### PR TITLE
Revision_interfaces

### PR DESCRIPTION
**Pull request by RubensGarofalo**

### CODE REVIEW

File: projects/pap/src/app/core/auth/auth.model.ts [hunk 1]
```diff
@@ -1,15 +1,62 @@
...
   zone_id: number;
 }
+
+export interface LoginCredentials {
+  email: string;
+  password: string;
+}
+
+export interface RegisterData extends User {
+  address: string;
```
Comment (peppedeka) [inline projects/pap/src/app/core/auth/auth.model.ts:49]: check if is a Address type

File: projects/pap/src/app/core/auth/auth.model.ts [hunk 2]
```diff
@@ -1,15 +1,62 @@
...
+  password: string;
+  password_confirmation: string;
+}
+
+export interface LogoutResponse extends User {
+  message: string;
+}
+
+export interface ResendEmailResponse {
+  data: any[];
```
Comment (webmappsrl) [inline projects/pap/src/app/core/auth/auth.model.ts:59]: check if is any[]

File: projects/pap/src/app/core/auth/auth.model.ts [hunk 3]
```diff
@@ -1,15 +1,62 @@
...
+  password: string;
+}
+
+export interface RegisterData extends User {
+  address: string;
+  password: string;
+  password_confirmation: string;
+}
+
+export interface LogoutResponse extends User {
```
Comment (webmappsrl) [inline projects/pap/src/app/core/auth/auth.model.ts:54]: check if is LogoutResponse or jush Response

File: projects/pap/src/app/features/calendar/calendar-page.component.ts [hunk 4]
```diff
@@ -53,15 +53,16 @@ export class CalendarPageComponent {
     this._inAppBrowser.create('https://www.esaspa.it/cittadini/calendario-pap');
   }
 
-  selectAddress(event: any): void {
+  selectAddress(event: Event): void {
```
Comment (webmappsrl) [inline projects/pap/src/app/features/calendar/calendar-page.component.ts:56]: check if is possible assign IonChangeEvent instead Event

File: projects/pap/src/app/features/calendar/calendar.model.ts [hunk 5]
```diff
@@ -7,8 +8,27 @@ export interface CalendarRow {
...
 
 export interface Calendar {
-  address: any;
+  address: Address;
   calendar: {
     [dateOfTrip: string]: CalendarRow[];
   };
 }
+
+export interface IonChangeEvent {
```
Comment (webmappsrl) [inline projects/pap/src/app/features/calendar/calendar.model.ts:17]: remove this interface, use CustomEvent<type_of_detail>

File: projects/pap/src/app/features/reports/reports.service.ts [hunk 6]
```diff
@@ -13,6 +13,6 @@ export class ReportsService {
   getReports(): Observable<Ticket[]> {
     return (this._http.get(`${env.api}/c/${env.companyId}/tickets`) as Observable<any>).pipe(
```
Comment (webmappsrl) [inline projects/pap/src/app/features/reports/reports.service.ts:14]: use a genericSuccess<T> interface

File: projects/pap/src/app/features/trash-book/trash-book-model.ts [hunk 7]
```diff
@@ -28,5 +31,13 @@ export interface TrashBookType {
   };
   slug: string;
   translations: any;
-  where: string;
+  where: {[lang: string]: string | null};
+}
+
+export interface IonSearchbarChangeEventDetail {
```
Comment (webmappsrl) [inline projects/pap/src/app/features/trash-book/trash-book-model.ts:37]: error

File: projects/pap/src/app/features/trash-book/trash-book-model.ts [hunk 8]
```diff
@@ -28,5 +31,13 @@ export interface TrashBookType {
...
+  value: string;
+}
+
+export interface IonSearchbarChangeEvent extends CustomEvent {
```
Comment (webmappsrl) [inline projects/pap/src/app/features/trash-book/trash-book-model.ts:41]: error use CustomEvent<string>

File: projects/pap/src/app/shared/form/model.ts [hunk 9]
```diff
@@ -47,3 +54,40 @@ export type TicketFieldTypes =
...
 export type TicketType = 'reservation' | 'info' | 'abandonment' | 'report';
+
+export interface ShowedIn {
+  abandonment: boolean;
+  info: boolean;
+  report: boolean;
+  reservation: boolean;
+}
+
+export interface Translation {
```
Comment (webmappsrl) [inline projects/pap/src/app/shared/form/model.ts:65]: remove if is useless

**Review by peppedeka — CHANGES_REQUESTED** [review]
resolve comments

**Pull request merged**